File: SGT.WebAdmin/ViewsScripts/Enumeradores/EnumRegimeLimpeza.js
var EnumRegimeLimpezaHelper = function () {
    this.Nenhum = 0;
    this.LimpezaASeco = 1;
    this.LimpezaComAgua = 2;
    this.LimpezaComAguaEAgenteDeLimpeza = 3;
    this.LimpezaEDesinfeccao = 4;
};

EnumRegimeLimpezaHelper.prototype = {
    obterOpcoes: function () {
        return [
            { text: "Nenhum", value: this.Nenhum },
            { text: "Regime A – Limpeza a seco", value: this.LimpezaASeco },
            { text: "Regime B – Limpeza com Água", value: this.LimpezaComAgua },
            { text: "Regime C – Limpeza com água e um agente de limpeza", value: this.LimpezaComAguaEAgenteDeLimpeza },
            { text: "Regime D – Limpeza e Desinfecção", value: this.LimpezaEDesinfeccao },
        ];
    },
}

var EnumRegimeLimpeza = Object.freeze(new EnumRegimeLimpezaHelper());

File: Dominio/ObjetosDeValor/Embarcador/Enumeradores/EnumRegimeLimpeza.cs
namespace Dominio.ObjetosDeValor.Embarcador.Enumeradores
{
    public enum EnumRegimeLimpeza
    {
        LimpezaASeco = 1,
        LimpezaComAgua = 2,
        LimpezaComAguaEAgenteDeLimpeza = 3,
        LimpezaEDesinfeccao = 4
    }

    public static class EnumRegimeLimpezaHelper
    {
        public static string ObterDescricao(this EnumRegimeLimpeza regime)
        {
            switch (regime)
            {
                case EnumRegimeLimpeza.LimpezaASeco: return "Regime A – Limpeza a seco";
                case EnumRegimeLimpeza.LimpezaComAgua: return "Regime B – Limpeza com Água";
                case EnumRegimeLimpeza.LimpezaComAguaEAgenteDeLimpeza: return "Regime C – Limpeza com água e um agente de limpeza";
                case EnumRegimeLimpeza.LimpezaEDesinfeccao: return "Regime D – Limpeza e Desinfecção";
                default: return string.Empty;
            }
        }
    }
}